source/distributed/tria.cc

```diff
@@ -3447,6 +3448,39 @@ namespace parallel
       update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim,spacedim>::repartition ()
+    {
+      AssertThrow(settings & no_automatic_repartitioning,
+                  ExcMessage("You need to set no_automatic_repartition when calling repartition() manually."));
+
```

Comment:
I think it would be useful if you added an assertion here that checks that no refinement/coarsening flags are set. I tend to think of this operation as not very different than actual mesh refinement and it might confuse people who think that this operation actually moves data around.